### PR TITLE
Restricting login OTP entries to 5 attempts #1

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/4514/limit-incorrect-otp-entries

### CODE REVIEW

**Review by msasikanth — CHANGES_REQUESTED** [review]
There seems to be a circular dagger dependency with the release build, please check.

File: app/src/main/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtection.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      maxAllowedFailedOtpAttempts: Int
+  ): BruteForceOtpProtectionState {
+    val otpState = otpStatePreference.get()
+    val isOtpAttemptLimitReached = otpState.failedLoginOtpAttempt >= maxAllowedFailedOtpAttempts
+    return if (isOtpAttemptLimitReached && !otpState.limitReachedAt.isPresent) {
+      otpState.failedAttemptLimitReached(utcClock)
+    } else {
+      otpState.loginAttemptFailed()
+    }
+  }
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtection.kt:28]: We can probably combine these 2 into 1 function? No need to do this now, you can do this in a different PR.

**Review by msasikanth — DISMISSED** [review]

File: app/src/main/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtection.kt (hunk 1 above)
Comment (Honey14) [reply app/src/main/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtection.kt:28]: Okay, adding this in a different PR

**Review by msasikanth — APPROVED** [review]
